src/components/TextField/TextArea.js

```diff
@@ -0,0 +1,276 @@
+    onKeyDown: PropTypes.func,
+    onKeyPress: PropTypes.func,
+    onKeyUp: PropTypes.func,
+    placeholder: PropTypes.string,
+    rows: PropTypes.number,
+    style: StyleObjectPropType(),
+    tooltipError: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    tooltipHint: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    tooltipRequired: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    value: PropTypes.string,
+};
+
+const defaultProps = {
+    className: null,
+    componentRef: null,
+    hasAutoHeight: false,
+    hasLabelAlways: false,
+    isDisabled: false,
+    isFullWidth: false,
+    isReadOnly: false,
+    isRequired: false,
+    isValid: null,
+    label: null,
+    name: null,
+    onBlur: null,
+    onChange: null,
+    onEnterKey: null,
+    onFocus: null,
+    onKeyDown: null,
+    onKeyPress: null,
+    onKeyUp: null,
+    placeholder: null,
+    rows: 1,
+    style: null,
+    tooltipError: null,
+    tooltipHint: null,
+    tooltipRequired: 'required',
+    value: '',
+};
+
+let lastInstanceId = 0;
+
+class TextArea extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hasFocus: false,
+            hasMouseOver: false,
+            showTooltip: false,
+            value: props.value,
+        };
+
+        lastInstanceId += 1;
+        this.uid = `textarea-${lastInstanceId}`;
+    }
+
+    componentDidMount = () => {
+        autosize(this.inputRef);
+    }
+
+    handleMouseEnter = () => {
+        this.setState({ hasMouseOver: true });
+    }
+
+    handleMouseLeave = () => {
+        this.setState({ hasMouseOver: false });
+    }
+
+    handleInputRef = (ref) => {
+        const { componentRef } = this.props;
+        this.inputRef = ref;
+        if (componentRef) {
+            componentRef(ref);
+        }
+    }
+
+    handleInputBlur = (event) => {
+        const onBlur = this.props.onBlur || (() => {});
+        this.setState({
+            hasFocus: false,
+            showTooltip: false,
+        }, () => {
+            onBlur(event);
+        });
+    }
+
+    handleInputChange = (event) => {
+        const { value } = this.inputRef;
+        const onChange = this.props.onChange || (() => {});
+        if (typeof value !== 'undefined') {
+            this.setState({ value }, () => {
+                onChange(value, event);
+            });
+        } else {
+            onChange(value, event);
+        }
+    }
+
+    handleInputFocus = (event) => {
+        const onFocus = this.props.onFocus || (() => {});
+        this.setState({
+            hasFocus: true,
+            showTooltip: true,
+        }, () => {
+            onFocus(event);
+        });
+    }
+
+    handleInputKeyDown = (event) => {
+        const { onKeyDown, onEnterKey } = this.props;
+        if (onKeyDown) {
+            onKeyDown(event.key, event);
```

Comment:
Handling errors from events is more important here. If `onKeyDown` errors, `onEnterKey` never gets called. That might be what we want, but it isn't obvious or documented from what I can see.